common/common-db/src/main/java/org/ow2/proactive/db/TransactionHelper.java

```diff
@@ -0,0 +1,123 @@
+package org.ow2.proactive.db;
+
+import org.objectweb.proactive.utils.Sleeper;
+import org.ow2.proactive.scheduler.core.properties.PASchedulerProperties;
+import org.apache.log4j.Logger;
+import org.hibernate.HibernateException;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.exception.LockAcquisitionException;
+
+
+public class TransactionHelper {
+
+    private static final Logger logger = Logger.getLogger(TransactionHelper.class);
+
+    private final SessionFactory sessionFactory;
+
+    public TransactionHelper(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    /**
+     * Execute the specified {@code sessionWork} in a session configured
+     * for read-only interactions with the database. This method should be
+     * used for select queries.
+     *
+     * @param sessionWork the action to perform.
+     * @param <T>         the result type of the session work.
+     * @return session work result based on the generic type.
+     */
+    public <T> T executeReadOnlyTransaction(SessionWork<T> sessionWork) {
+        return tryExecuteTransaction(sessionWork, false, true);
+    }
+
+    /**
+     * Execute the specified {@code sessionWork} in a session configured
+     * for read/write interactions with the database. This method should be
+     * used for insert, update, and delete queries. If select only is required,
+     * then {@link #executeReadOnlyTransaction(SessionWork)} could be used.
+     * <p>
+     * In case of database error while executing the query, a rollback is
+     * performed.
+     *
+     * @param sessionWork the action to perform.
+     * @param <T>         the result type of the session work.
+     * @return session work result based on the generic type.
+     */
+    public <T> T executeReadWriteTransaction(SessionWork<T> sessionWork) {
```

Comment:
I understand that the read-only transaction changes the behavior of commit/rollback semantics, but I don't understand why it changes the retry mechanism.

Don't we want that if there is a database failure (could be database server network disconnection), we try to recover automatically without impacting the scheduler ? In both read and write operations ?